src/cargo/core/compiler/mod.rs

```diff
@@ -1123,7 +1076,8 @@ struct OutputOptions {
 
 impl OutputOptions {
     fn new<'a>(cx: &Context<'a, '_>, unit: &Unit<'a>) -> OutputOptions {
-        let look_for_metadata_directive = cx.rmeta_required(unit);
+        let look_for_metadata_directive =
+            cx.rmeta_required(unit) || cx.bcx.config.cli_unstable().timings.is_some();
```

Comment:
Could the calculation for `rmeta_required` take the clause for timing into account? I think that' remove the need to update `add_error_format_and_color` as well and at least at a cursory glance it seems like it'd be fine to change that definition